src/Adapters/BlobAdapter.ts

```diff
@@ -36,16 +44,50 @@ export default class BlobAdapter implements IBlobAdapter {
     protected containerResourceId: string; // resource scope
     protected blobAuthService: IAuthService;
     protected blobProxyServerPath: string;
+    protected useBlobProxy: boolean;
 
     constructor(
         blobContainerUrl: string,
         authService: IAuthService,
-        blobProxyServerPath = '/proxy/blob'
+        blobProxyServerPath = '/proxy/blob',
+        useBlobProxy = true
     ) {
         this.setBlobContainerPath(blobContainerUrl);
         this.blobAuthService = authService;
         this.blobAuthService.login();
         this.blobProxyServerPath = blobProxyServerPath;
+        /**
+         * Check if class has been initialized with CORS enabled or if origin matches dev or prod explorer urls,
+         * override if CORS is forced by feature flag
+         *  */
+        this.useBlobProxy =
+            (useBlobProxy || !validateExplorerOrigin(window.origin)) &&
+            !forceCORS;
+    }
+
+    getBlobContainerURL() {
+        return this.storageAccountHostName && this.containerName
+            ? `https://${this.storageAccountHostName}/${this.containerName}`
+            : '';
+    }
+
+    generateBlobUrl(path: string) {
```

Comment:
nit: can we change the name to reflect the argument?
```suggestion
    generateBlobUrlWithPath(path: string) {
```